### PR TITLE
 mgr/dashboard: Allow renaming an existing Pool

### PR DESCRIPTION
**Pull request by familyuu**
mgr/dashboard:  Allow renaming an existing Pool

Fixes: http://tracker.ceph.com/issues/36560

Signed-off-by: familyuu guodan1@lenovo.com

### CODE REVIEW

File: src/pybind/mgr/dashboard/controllers/pool.py [hunk 1]
```diff
@@ -117,6 +117,8 @@ def set_key(key, value):
             CephService.send_command('mon', 'osd pool set', pool=pool, var=key, val=str(value))
 
         for key, value in kwargs.items():
+            if key == 'pool':
+               CephService.send_command('mon', 'osd pool rename', srcpool=pool, destpool=value)
```
Comment (tchaikov) [inline src/pybind/mgr/dashboard/controllers/pool.py]:
```
************* Module dashboard.controllers.pool
W:121, 0: Bad indentation. Found 15 spaces, expected 16 (bad-indentation)
```

@familyuu might want to address this issue.

Comment (familyuu) [reply src/pybind/mgr/dashboard/controllers/pool.py]: Yeah, I really want to solve this issue, but something is not completed, I have always working on it for to solve it. this PR may be submitted too anxious.

Comment (LenzGr) [reply src/pybind/mgr/dashboard/controllers/pool.py]: > this PR may be submitted too anxious

No problem, just keep on evolving it!

**Comment by votdev** [comment]
Doesn't work for me.
error

**Comment by familyuu** [comment]
@votdev 
ezgif-1-d834604c6057

**Comment by votdev** [comment]
@familyuu I use a vstart cluster in our [ceph-dev-docker](https://github.com/ricardoasmarques/ceph-dev-docker) environment. Ceph was build from master on the same day.

**Comment by votdev** [comment]
@familyuu I use a vstart cluster in our [ceph-dev-docker](https://github.com/ricardoasmarques/ceph-dev-docker) environment. Ceph was build from master on the same day.

**Comment by familyuu** [comment]
> @familyuu I use a vstart cluster in our [ceph-dev-docker](https://github.com/ricardoasmarques/ceph-dev-docker) environment. Ceph was build from master on the same day.
Pls try again, I found out error and fixed it.

**Comment by callithea** [comment]
jenkins restest this please

**Comment by callithea** [comment]
> @votdev
> Pls try again, I found out error and fixed it.
> The error you found is because I send the 'rename' command to ceph too earlier, result some other command that need the pool name is failed with 'unrecognized pool somePoolName'.
> Thanks a lot for your feedback.

I tested your PR locally and it works for me to rename a pool! :)

**Comment by tchaikov** [comment]
retest this please.

File: src/pybind/mgr/dashboard/controllers/pool.py [hunk 2]
```diff
@@ -117,9 +118,15 @@ def set_key(key, value):
...
-                set_key('pgp_num', value)
+            if key == 'pool':
+               update_name = True
+               destpool = value
+            else:
+               set_key(key, value)
+               if key == 'pg_num':
+                  set_key('pgp_num', value)
+        if  update_name:
+            CephService.send_command('mon', 'osd pool rename', srcpool=pool, destpool=destpool)
```
Comment (tchaikov) [inline src/pybind/mgr/dashboard/controllers/pool.py]:
```
************* Module dashboard.controllers.pool
W:122, 0: Bad indentation. Found 15 spaces, expected 16 (bad-indentation)
W:123, 0: Bad indentation. Found 15 spaces, expected 16 (bad-indentation)
W:125, 0: Bad indentation. Found 15 spaces, expected 16 (bad-indentation)
W:126, 0: Bad indentation. Found 15 spaces, expected 16 (bad-indentation)
W:127, 0: Bad indentation. Found 18 spaces, expected 20 (bad-indentation)
```

could you fix the indent?

File: src/pybind/mgr/dashboard/controllers/pool.py [hunk 3]
```diff
@@ -117,9 +118,15 @@ def set_key(key, value):
...
-            if key == 'pg_num':
-                set_key('pgp_num', value)
+            if key == 'pool':
+               update_name = True
+               destpool = value
+            else:
+               set_key(key, value)
+               if key == 'pg_num':
+                  set_key('pgp_num', value)
+        if  update_name:
```
Comment (tchaikov) [inline src/pybind/mgr/dashboard/controllers/pool.py]: might want to remove the duplicated space after `if`.

**Comment by LenzGr** [comment]
@familyuu - thanks again for your contribution. Could you please rebase your code and resolve the remaining issues? The `run-tox.sh` script currently reports the following warning about the Python code:
```
************* Module dashboard.controllers.pool
W:122, 0: Bad indentation. Found 15 spaces, expected 16 (bad-indentation)
W:123, 0: Bad indentation. Found 15 spaces, expected 16 (bad-indentation)
W:125, 0: Bad indentation. Found 15 spaces, expected 16 (bad-indentation)
W:126, 0: Bad indentation. Found 15 spaces, expected 16 (bad-indentation)
W:127, 0: Bad indentation. Found 18 spaces, expected 20 (bad-indentation)
```

**Comment by familyuu** [comment]
> @familyuu - thanks again for your contribution. Could you please rebase your code and resolve the remaining issues? The `run-tox.sh` script currently reports the following warning about the Python code:
> 
> ```
> ************* Module dashboard.controllers.pool
> W:122, 0: Bad indentation. Found 15 spaces, expected 16 (bad-indentation)
> W:123, 0: Bad indentation. Found 15 spaces, expected 16 (bad-indentation)
> W:125, 0: Bad indentation. Found 15 spaces, expected 16 (bad-indentation)
> W:126, 0: Bad indentation. Found 15 spaces, expected 16 (bad-indentation)
> W:127, 0: Bad indentation. Found 18 spaces, expected 20 (bad-indentation)
> ``

Changed, and jenkins test failed with error: Agent is offline.

**Comment by callithea** [comment]
jenkins test dashboard

**Pull request merged**